code/datums/elements/connect_loc.dm

```diff
@@ -24,25 +24,23 @@
 
 	src.connections = connections
 
-	RegisterSignal(tracked, COMSIG_MOVABLE_LOCATION_CHANGE, .proc/on_moved)
+	RegisterSignal(tracked, COMSIG_MOVABLE_LOCATION_CHANGE, .proc/on_moved, override=TRUE)
 	update_signals(listener, tracked)
 
 /datum/element/connect_loc/Detach(datum/listener, atom/movable/tracked, list/connections)
 	. = ..()
 
 	if(!tracked)
-		tracked = listener
-
-	if(!istype(tracked))
-		return
-
-	unregister_signals(listener, tracked, tracked.loc)
-
-	UnregisterSignal(tracked, COMSIG_MOVABLE_LOCATION_CHANGE)
+		unregister_all(listener)
+	else if(targets[tracked.loc]) // Detach can happen multiple times due to qdel
+		unregister_signals(listener, tracked, tracked.loc)
+		UnregisterSignal(tracked, COMSIG_MOVABLE_MOVED)
```

Comment:
This is dead code isn't it? since the line above it should always be run by Destroy()?

Reply:
No, Detach can run without Destroy, and it needs to clean itself up properly.

Reply:
Oh wait I'm dumb, you might be right.

Reply:
it's dead in that nothing currently uses it, but it could be used in theory so it's not kill